command/eval_delete.go

```diff
@@ -0,0 +1,529 @@
+
+	// originalBrokerPaused indicates whether the broker was in a paused state
+	// before the command was run. This indicates what action, if any, should
+	// be taken before the command finishes.
+	originalBrokerPaused bool
+
+	// deleteByArg is set when the command is deleting an evaluation that has
+	// been passed as an argument. This avoids need for confirmation.
+	deleteByArg bool
+
+	// numDeleted tracks the total evaluations deleted in a single run of this
+	// command. It provides a way to output this information to the user at the
+	// command completion.
+	numDeleted int
+
+	// client is the lazy-loaded API client and is stored here, so we don't
+	// need to pass it to multiple functions.
+	client *api.Client
+}
+
+func (e *EvalDeleteCommand) Help() string {
+	helpText := `
+Usage: nomad eval delete [options] <evaluation>
+
+  Delete an evaluation by ID. If the evaluation ID is omitted, this command
+  will use the filter options below to delete a set of evaluations. If ACLs
+  are enabled, this command requires a management ACL token.
+
+  This command should be used cautiously and only in outage situations where
+  there is a large backlog of evaluations not being processed. During most
+  normal and outage scenarios, Nomads reconciliation and state management will
+  handle evaluations as needed.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsNoNamespace) + `
+
+Eval Delete Options:
+
+  -scheduler
+    Filter by scheduler type (one of "batch", "service", "system", or
+    "sysbatch"). This option may be specified multiple times.
+
+  -status
+    Filter by eval status (one of "blocked", "pending", "complete",
+    "failed", or "canceled"). This option may be specified multiple times.
+
+  -job
+    Filter by an associated job ID. This option may be specified
+    multiple times.
+
+  -deployment
+    Filter by an associated deployment ID. This option may be specified
+    multiple times.
+
+  -node
+    Filter by an associated node ID. This option may be specified
+    multiple times.
+
+  -filter-operator
+    The filter operator to use when multiple filter options are specified. This
+    defaults to "and" but also supports "or".
+
+  -yes
+    Bypass the confirmation prompt if an evaluation ID was not provided.
+`
+
+	return strings.TrimSpace(helpText)
+}
+
+func (e *EvalDeleteCommand) Synopsis() string {
+	return "Delete evaluations by ID or using filters"
+}
+
+func (e *EvalDeleteCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(e.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-scheduler": complete.PredictSet(
+				api.JobTypeService,
+				api.JobTypeBatch,
+				api.JobTypeSystem,
+				api.JobTypeSysBatch,
+			),
+			"-status":          complete.PredictAnything,
+			"-job":             complete.PredictAnything,
+			"-deployment":      complete.PredictAnything,
+			"-node":            complete.PredictAnything,
+			"-filter-operator": complete.PredictSet("and", "or"),
```

Comment:
> Rather than expose operators and HashiCorp field teams to an unsafely sharp tool, the initial implementation of this command will require exercising fine-grained control over which evaluations to delete

I went down this road according to what was written and approved within the RFC document. If we now prefer to go down the route which keeps consistency with other commands I can go ahead and pull this out and implement the single filter flag. This at least had an added benefit that operator can test the filter via the list command first, before running it on the delete.

Reply:
This change has been made within commit https://github.com/hashicorp/nomad/pull/13492/commits/1ef732e2e96a690cda3ed14737abbaa88836ddbd

Reply:
Ugh sorry for missing this in RFC review James. :(